src/CommandAction.ts

```diff
@@ -42,15 +42,12 @@ export default class CommandAction implements Action {
 
           editor.selections = targetSelections;
 
-          const promise = listenForDocumentChanges(editor, [
-            originalSelections,
-            targetSelections,
-          ]);
-
-          await commands.executeCommand(this.command);
-
           const [updatedOriginalSelections, updatedTargetSelections] =
-            await promise;
+            await callFunctionAndUpdateSelections(
+              editor,
+              () => commands.executeCommand(this.command),
+              [originalSelections, targetSelections]
+            );
```

Comment:
WDYT of this one?  I didn't implement it but it should be trivial given what you've already built

Reply:
Unsure. The function itself should work but I don't know if we're going to use it any in any more places. If it is only used in CommandAction I would say that we don't need it.